### PR TITLE
Add Simplex::ScalarPolynomial

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/deal.II/simplex/polynomials.h [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+namespace Simplex
+{
+  /**
+   * Polynomials defined on dim-dimensional simplex entities. This class is
+   * basis of Simplex::FE_P.
+   */
+  template <int dim>
+  class ScalarPolynomial : public ScalarPolynomialsBase<dim>
+  {
+    static_assert(dim == 2 || dim == 3, "Dimension not supported!");
```
Comment (luca-heltai) [inline include/deal.II/simplex/polynomials.h:39]: Why is dim == 1 not supported?

**Comment by peterrum** [comment]
> In what is this different from PolynomialsP?

PolynomialsP limits the polynomial order. E.g.: `PolynomialsP<2>()` has following shape functions `1`, `x`, `y` but not `xy`.

My class puts the support points into the vertices, middle of lines/faces.

File: source/simplex/polynomials.cc [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+
+      return 0;
+    }
+  } // namespace
+
+  template <int dim>
+  ScalarPolynomial<dim>::ScalarPolynomial(const unsigned int degree)
+    : ScalarPolynomialsBase<dim>(degree, compute_n_polynomials(dim, degree))
+  {}
+  template <int dim>
```
Comment (drwells) [inline source/simplex/polynomials.cc:53]:
```suggestion



  template <int dim>
```

File: source/simplex/polynomials.cc [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+            return 10;
+        }
+
+      Assert(false, ExcNotImplemented());
+
+      return 0;
+    }
+  } // namespace
+
+  template <int dim>
```
Comment (drwells) [inline source/simplex/polynomials.cc:49]:
```suggestion


  template <int dim>
```

Comment (drwells) [reply source/simplex/polynomials.cc:49]: We need three blank lines between function definitions - please check the rest.

**Comment by peterrum** [comment]
@drwells Done!

**Pull request merged**